js/ui/scheduler/appointmentPopup.js

```diff
@@ -211,28 +211,14 @@ export default class AppointmentPopup {
             this.scheduler.fire('setField', 'endDate', formData, endDate);
         }
 
-        const { startDateExpr, endDateExpr, recurrenceRuleExpr } = this.scheduler._dataAccessors.expr;
-
-        const recurrenceEditorOptions = this._getEditorOptions(recurrenceRuleExpr, APPOINTMENT_FORM_GROUP_NAMES.Recurrence);
-        this._setEditorOptions(recurrenceRuleExpr, APPOINTMENT_FORM_GROUP_NAMES.Recurrence, extend({}, recurrenceEditorOptions, { startDate: startDate }));
+        const { startDateExpr, endDateExpr } = this.scheduler._dataAccessors.expr;
 
         this._appointmentForm.option('readOnly', this._isReadOnly(data));
 
-        AppointmentForm.updateFormData(this._appointmentForm, formData);
+        AppointmentForm.updateFormData(this._appointmentForm, formData, this.scheduler._dataAccessors.expr, startDate, endDate);
```

Comment:
A lot of arguments.
Can we use `this.scheduler._dataAccessors.expr` inside `updateFormData` method?
Does `formData` not contains `startDate`, `endDate`?

Reply:
@AntonSermyazhko There are some difficulties with refactoring there. ui.scheduler.appointmentForm is a kind of facade above Form.  This entity does not have access to compiled getters to access fields. But I will try to refactor